TombEngine/Objects/TR4/Trap/tr4_sethblade.cpp

```diff
@@ -1,83 +1,86 @@
 #include "framework.h"
-#include "tr4_sethblade.h"
-#include "Specific/level.h"
-#include "Specific/setup.h"
-#include "Game/control/control.h"
+#include "Objects/TR4/Trap/tr4_sethblade.h"
+
 #include "Game/animation.h"
+#include "Game/control/control.h"
 #include "Game/items.h"
+#include "Specific/level.h"
+#include "Specific/setup.h"
 
 namespace TEN::Entities::TR4
 {
+	constexpr auto SETH_BLADE_HARM_DAMAGE = 1000;
+
 	enum SethBladeState 
 	{
-		SETHBLADE_STATE_NONE = 0,
+		// No state 0.
 		SETHBLADE_STATE_ACTIVE = 1,
-		SETHBLADE_STATE_IDLE = 2,
+		SETHBLADE_STATE_IDLE = 2
 	};
 
 	enum SethBladeAnim 
 	{
 		SETHBLADE_ANIM_ACTIVATE = 0,
-		SETHBLADE_ANIM_IDLE = 1,
+		SETHBLADE_ANIM_IDLE = 1
 	};
 
 	void InitialiseSethBlade(short itemNumber)
 	{
-		auto* item = &g_Level.Items[itemNumber];
+		auto& item = g_Level.Items[itemNumber];
 
-		item->Animation.AnimNumber = Objects[item->ObjectNumber].animIndex + SETHBLADE_ANIM_IDLE;
-		item->Animation.TargetState = SETHBLADE_STATE_ACTIVE;
-		item->Animation.ActiveState = SETHBLADE_STATE_IDLE;
+		SetAnimation(&item, SETHBLADE_ANIM_IDLE);
+		item.ItemFlags[2] = abs(item.TriggerFlags); // NOTE: ItemFlags[2] stores blade timer.
+		item.ItemFlags[3] = SETH_BLADE_HARM_DAMAGE; // NOTE: ItemFlags[3] stored blade harm damage.
 
-		item->Animation.FrameNumber = g_Level.Anims[item->Animation.AnimNumber].frameBase;
-		item->ItemFlags[2] = abs(item->TriggerFlags);
+		// Immediately start blades.
```

Comment:
I don't understand the purpose of this addition. First, you copy `abs(item.TriggerFlags)` to `item.ItemFlags[2]`, but here you immediately overwrite it with 1. Why?

Reply:
Basically.

If the OCB is 0, it goes to 1 to trigger the activation state.
If the OCB is >0, it was doing a waiting, then it does the activation, and repeat the waiting. Initializing the >0 cases, makes the blade to get activated when it's triggered, and then keeps repeating with the waiting of frames indicated in the OCB.

If the OCB is <0, this is the special case used in SethTomb where the blades do a wait of frames before to trigger, so here yes it's initialized with abs(OCB), (in editor is put in negative but we just want positive values for the counter).

Still, now you mentioned this, I though it could be better written in 1 line, I'm updatting it with the next commit.